app/models/report.rb

```diff
@@ -3,7 +3,16 @@
 class Report < ApplicationRecord
   belongs_to :user
   has_many :comments, as: :commentable, dependent: :destroy
-
+  has_many :mentioning_relationships, class_name: 'Mention',
+                                      foreign_key: 'mentioning_id',
+                                      dependent: :destroy,
+                                      inverse_of: :mentioning
+  has_many :mentioning_reports, through: :mentioning_relationships, source: :mentioned
+  has_many :mentioned_relationships,  class_name: 'Mention',
+                                      foreign_key: 'mentioned_id',
+                                      dependent: :destroy,
+                                      inverse_of: :mentioned
+  has_many :mentioned_reports, through: :mentioned_relationships, source: :mentioning
```

Comment:
```suggestion
  has_many :mentioned_reports, through: :mentioned_relationships, source: :mentioning

```

has_manyとvalidatesの間は空行を入れることが多いかも